RFCs/RFC-0: Lightweight RFC process.md

```diff
@@ -0,0 +1,78 @@
+# RFC-0: Lightweight RFC Process for riff
```

Comment:
I have a personal preference for lowercase directories and filenames, but will follow the group consensus

Reply:
+1 for lowercase filenames too, with dashes as separators when it makes sense